packages/dom-helper/lib/main.js

```diff
@@ -565,15 +567,39 @@ prototype.parseHTML = function(html, contextualElement) {
 };
 
 var parsingNode;
+var URL;
 
-// Used to determine whether a URL needs to be sanitized.
 prototype.protocolForURL = function(url) {
+  // Test to see if our DOM implementation parses
+  // and normalizes URLs.
+  var protocol = browserProtocolForURL.call(this, 'foobar:baz');
+  if (protocol === 'foobar:') {
+    // Swap in the method that doesn't do this test now that
+    // we know it works.
+    prototype.protocolForURL = browserProtocolForURL;
+  } else {
+    // Otherwise, we need to fall back to our own URL parsing.
+    // Global `require` is shadowed by Ember's loader so we have to use the fully
+    // qualified `module.require`.
+    URL = module.require('url');
```

Comment:
what if there is no require? Might be nice to fail fast then (might be aligned with my comment bellow about doing this before setting up the prototype.